test262/per-file/main.js

```diff
@@ -0,0 +1,225 @@
+      const childNode = template.content.children[0].cloneNode(true);
+      node.appendChild(childNode);
+      childNode.querySelector("#name").textContent = childName;
+      childNode.querySelector("#status").innerHTML = generateStatus(
+        child.aggregatedResults
+      );
+      childNode.querySelector("#url").href =
+        "https://github.com/tc39/test262/tree/main/" +
+        pathInTree.join("/") +
+        "/" +
+        childName;
+      if (!isLeaf) {
+        childNode
+          .querySelectorAll(".tree-node-name, .tree-node-image")
+          .forEach((childNode) => {
+            childNode.onclick = function () {
+              pathInTree.push(childName);
+              generateChildren(node);
+            };
+          });
+      }
+    });
+}
+
+function color(name) {
+  return TestResultColors[name] || "black";
+}
+
+function resultAwareSort(names) {
+  const resultOrder = [
+    TestResult.PASSED,
+    TestResult.FAILED,
+    TestResult.SKIPPED,
+    TestResult.PROCESS_ERROR,
+    TestResult.TODO_ERROR,
+    TestResult.METADATA_ERROR,
+    TestResult.HARNESS_ERROR,
+    TestResult.TIMEOUT_ERROR,
+    TestResult.RUNNER_EXCEPTION,
+    TestResult.DURATION,
+  ];
+
+  return names.sort((a, b) => {
+    const aIndex = resultOrder.indexOf(a);
+    const bIndex = resultOrder.indexOf(b);
+    return aIndex - bIndex;
+  });
+}
+
+function generateStatus(aggregatedResults) {
+  const status = Object.keys(aggregatedResults)
+    .sort()
+    .reduce((acc, mode) => {
+      const stats = aggregatedResults[mode];
+      const total = Object.keys(stats).reduce(
+        (acc, name) => acc + stats[name],
+        0
+      );
+      if (total === 0) return acc;
+      acc.push(`<div class="mode-summary-container">
+        <span class="mode-result-text">${mode}</span>
+        <div class="mode-bar-container">
+            ${resultAwareSort(Object.keys(stats))
+              .map(
+                (x) =>
+                  `<div title="${
+                    TestResultLabels[x] +
+                    ": " +
+                    ((100 * stats[x]) / total).toFixed(2) +
+                    "%"
+                  }" style="width: ${
+                    (100 * stats[x]) / total
+                  }%; height: 20px; background-color: ${color(x)}"></div>`
+              )
+              .join("")}
+        </div>
+    </div>`);
+      return acc;
+    }, []);
+  return status.join(" ");
+}
+document.addEventListener("DOMContentLoaded", () => {
+  const headers = new Headers();
+  headers.append("pragma", "no-cache");
+  headers.append("cache-control", "no-cache");
+  const promises = [];
+  for (const mode of ["", "-bytecode"]) {
+    promises.push(
+      fetch(new Request(`../data/per-file${mode}-master.json`), {
+        method: "GET",
+        headers,
+      })
+        .then((response) => response.json())
+        .then((data) => initialize(data, mode))
+    );
+  }
```

Comment:
Is it overkill to consolidate this and the one on the other page into something like `fetchData("filename.json")` in a `utils.js`?

Reply:
shure, will do